public/controllers.js

```diff
@@ -436,3 +441,125 @@ appCtrls.controller('DevicesController', function DevicesController($scope, $int
 
   init();
 });
+
+appCtrls.controller('DeviceModalController', function DeviceModalController($scope, $rootScope, $uibModalInstance, GatewayApi) {
+  $scope.title = 'Create a new device';
+  $scope.serviceForm = {
+    thingd: {
+      dataItems: []
+    }
+  };
+  $scope.currentStep = 'CREATE_DEVICE';
+  $scope.form = {
+    thingd: {
+      name: '',
+      modbusSlaveID: undefined,
+      modbusSlaveURL: '',
+      dataItems: []
+    },
+    dataItem: {
+      schema: {
+        sensorID: undefined,
+        sensorName: '',
+        typeID: '',
+        unit: '',
+        valueType: ''
+      },
+      modbus: {
+        registerAddress: undefined,
+        bitOffset: '1'
+      },
+      config: {}
+    },
+    config: {
+      lowerThreshold: undefined,
+      upperThreshold: undefined,
+      change: 'yes',
+      timeSec: undefined
+    }
+  };
+
+  function clearDataItemForm() {
+    $scope.form.dataItem = {
+      schema: {
+        sensorID: undefined,
+        sensorName: '',
+        typeID: '',
+        unit: '',
+        valueType: ''
+      },
+      modbus: {
+        registerAddress: undefined,
+        bitOffset: '1'
+      },
+      config: {}
+    };
+    $scope.form.config = {
+      lowerThreshold: undefined,
+      upperThreshold: undefined,
+      change: 'yes',
+      timeSec: undefined
+    };
+  }
```

Comment:
Couldn't we reuse the `$scope.form.dataItem` (maybe creating an `emptyForm` there) defined above to avoid repetition?

Reply:
Done!